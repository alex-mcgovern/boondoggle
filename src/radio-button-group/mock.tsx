import type * as React from "react";

import type { RadioButton } from "../radio-button";

import { LOREM } from "../../mocks/LOREM.mock";

export const RADIO_BUTTONS_MOCK: Array<
	React.ComponentProps<typeof RadioButton>
> = [
	{
		description: LOREM.text_xxs,
		name: "radio_item_1",
		title: "Radio item 1",
		value: "radio_item_1",
	},
	{
		description: LOREM.text_xxs,
		name: "radio_item_2",
		title: "Radio item 2",
		value: "radio_item_2",
	},
	{
		description: LOREM.text_xxs,
		name: "radio_item_3",
		title: "Radio item 3",
		value: "radio_item_3",
	},
];
