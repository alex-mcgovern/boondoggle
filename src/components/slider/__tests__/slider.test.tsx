/** @jest-environment jsdom */
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { Slider } from "..";
import { LOREM } from "../../../../mocks/LOREM.mock";
import "../../../../test/has_pointer_capture.mock";
import "../../../../test/resize_observer.mock";

import type { SliderProps } from "..";

const PROPS: SliderProps = { name: LOREM.name() };

const renderComponent = ({ ...props }: SliderProps) => {
  return render(<Slider {...props} />);
};

describe("<Slider />", () => {
  describe("Basic smoke tests", () => {
    it("should render without throwing", () => {
      const { container } = renderComponent(PROPS);

      expect(container).not.toBeNull();
    });

    it("should match snapshot", () => {
      const { container } = renderComponent(PROPS);

      expect(container).toMatchSnapshot();
    });
  });

  describe("labelling", () => {
    /** RTL doesn't see this element ¯\_(ツ)_/¯ */
    it.skip("should assign name to the hidden input element", () => {
      const { getByRole } = renderComponent({
        name: LOREM.name(),
        placeholder: LOREM.placeholder(),
      });
      expect(
        (getByRole("textbox", { hidden: true }) as HTMLInputElement).name
      ).toBe(LOREM.name());
    });

    it("should label element correctly", () => {
      const { getByLabelText } = renderComponent({
        name: LOREM.name(),
        placeholder: LOREM.placeholder(),
        label: LOREM.label(),
        id: LOREM.id(),
      });

      expect(getByLabelText(LOREM.label())).not.toBeNull();
    });
  });

  describe("setting new value", () => {
    it("should update value when user interacts", async () => {
      const onValueChange = jest.fn();

      const { getByRole } = renderComponent({
        name: LOREM.name(),
        placeholder: LOREM.placeholder(),
        onValueChange,
      });

      const sliderThumb = getByRole("slider");
      await userEvent.click(sliderThumb);
      await userEvent.keyboard("{arrowright}");

      expect(onValueChange).toBeCalledWith([1]);
    });
  });
});
