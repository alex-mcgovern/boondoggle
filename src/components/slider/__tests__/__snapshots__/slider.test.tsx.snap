// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> Basic smoke tests should match snapshot 1`] = `
<div>
  <div
    class="get_sprinkles_color_text_low_contrast__eiij7b1k"
  >
    <div
      class="get_sprinkles_alignItems_center__eiij7b2 get_sprinkles_display_flex_mobile__eiij7b8d get_sprinkles_gap_spacing_1_mobile__eiij7b9m"
    >
      <span
        aria-disabled="false"
        aria-label="Name ipsum"
        class="slider.styles_sliderRoot__1aufb7k1 slider.styles__1aufb7k0 get_sprinkles_marginTop_spacing_1__eiij7b4w get_sprinkles_marginBottom_spacing_1__eiij7b42 get_sprinkles_alignItems_center__eiij7b2 get_sprinkles_display_flex_mobile__eiij7b8d get_sprinkles_position_relative_mobile__eiij7bfg get_sprinkles_width_100%_mobile__eiij7bfy"
        data-orientation="horizontal"
        dir="ltr"
        style="--radix-slider-thumb-transform: translateX(-50%);"
      >
        <span
          class="slider.styles_sliderTrack__1aufb7k3 slider.styles__1aufb7k2 get_sprinkles_borderRadius_sm__eiij7b10 get_sprinkles_position_relative_mobile__eiij7bfg"
          data-orientation="horizontal"
        >
          <span
            class="slider.styles_sliderRange__1aufb7k5 slider.styles__1aufb7k4 get_sprinkles_background_button_default__eiij7bf get_sprinkles_borderRadius_pill__eiij7bz get_sprinkles_height_100%__eiij7b2q get_sprinkles_position_absolute_mobile__eiij7bfj"
            data-orientation="horizontal"
            style="left: 0%; right: 100%;"
          />
        </span>
        <span
          style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
        >
          <span
            aria-orientation="horizontal"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="0"
            class="slider.styles_sliderThumb__1aufb7k7 slider.styles__1aufb7k6 get_sprinkles_background_tint_default__eiij7bm get_sprinkles_border_border_active__eiij7bq get_sprinkles_borderRadius_50%__eiij7bw get_sprinkles_display_block_mobile__eiij7b8a get_sprinkles_height_spacing_3__eiij7b2z get_sprinkles_width_spacing_3_mobile__eiij7bhp"
            data-orientation="horizontal"
            data-radix-collection-item=""
            role="slider"
            style=""
            tabindex="0"
          />
        </span>
      </span>
    </div>
  </div>
</div>
`;
